crm_timesheet/models/crm_phonecall.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# See README.rst file on addon root folder for license details
+
+from openerp import models, fields, api, _
+from openerp.fields import DATE_LENGTH
+from datetime import datetime
+from openerp.exceptions import ValidationError
+
+
+class CrmPhonecall(models.Model):
+    _inherit = 'crm.phonecall'
+
+    analytic_account_id = fields.Many2one(
+        comodel_name='account.analytic.account')
+    timesheet_ids = fields.One2many(comodel_name='hr.analytic.timesheet',
+                                    inverse_name='phonecall_id')
+
+    def _timesheet_prepare(self, vals):
+        """
+            param: vals
+            type: dict
+            desc: crm.phonecall vals dict (It's like create and write methods)
+        """
+        if len(self) > 0:
+            self.ensure_one()
+        date = vals.get('date', self.date)
+        user_id = vals.get('user_id', getattr(self.user_id, 'id', False))
+        account_id = vals.get('analytic_account_id',
+                              getattr(self.analytic_account_id, 'id', False))
+        unit_amount = vals.get('duration', self.duration)
+        res = {
+            'date': date[:DATE_LENGTH] if date else False,
+            'user_id': user_id,
+            'name': vals.get('name', self.name),
+            'account_id': account_id,
+            'unit_amount': unit_amount / 60.0,
+            'code': 'phone',
+            'journal_id': self.env[
+                'hr.analytic.timesheet']._getAnalyticJournal()
+        }
+        return res
+
+    @api.model
+    def create(self, vals):
+        if vals.get('analytic_account_id') and vals.get('duration', 0) > 0:
+            if not vals.get('date'):
+                raise ValidationError(_('Date field must be filled.'))
+            timesheet_data = self._timesheet_prepare(vals)
+            vals['timesheet_ids'] = vals.get('timesheet_ids', [])
+            vals['timesheet_ids'].append((0, 0, timesheet_data))
+        res = super(CrmPhonecall, self).create(vals)
+        return res
+
+    @api.one
```

Comment:
The same disertation about mass-writing records. In this case, it's more difficult to handle an scenario with multiple records, so maybe you can stay as is, but mention it on the README in known issues.